### DISCUSSION TITLE
Folder structure discussion

### DISCUSSION THREAD

**Discussion by shubham-kv** [ademclk/callofcode#59] [General]
At the time of writing this, the folder structure seems all messed up with c, cpp, java, etc. files all in one folder.

I'd like to propose for a better folder structure where the major folders (i.e. data_structures, algorithms, etc.) are grouped according to the language used. For example, we can have a C subfolder in algorithms, where the algorithms are implemented in C,
a C++ subfolder for all the algorithms implemented in C++ and so on...

Something like this:
<pre>
<code>
  /algorithms
  ../C
  ..../bubble_sort
  ..../insertion_sort
  ../C++
  ..../bubble_sort
  ..../insertion_sort

  /data_structures
  ../C
  ..../singly_linked_list
  ..../doubly_linked_list
  ../C++
  ..../singly_linked_list
  ..../doubly_linked_list
</code>
</pre>

This would help while navigating through the project and would make the project more clean.
What are your opinions ?

**Comment by shubham-kv** [comment]
I know it's not a major concern but it's good enough for a first discussion.

**Comment by ademclk** [comment]
I liked your idea. I'll change folder structure now. Ty for your opinion.
